### PR TITLE
python3.10 version bump

### PR DESCRIPTION
**Pull request by bpblanken**

### CODE REVIEW

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.8]
+        python-version: ['3.10']
```
Comment (bpblanken) [inline .github/workflows/unit-tests.yml:17]: bug in GitHub actions requiring a string [here](https://github.com/actions/runner/issues/1989)

File: requirements.in [hunk 2]
```diff
@@ -1,5 +1,5 @@
 elasticsearch==7.9.1
 google-api-python-client>=1.8.0
-hail>=0.2.113
-luigi==2.8.11
+hail==0.2.122
+luigi>=3.4.0
```
Comment (bpblanken) [inline requirements.in:4]: `luigi 2.8.11` didn't work with 3.10, so I bumped.  Seems to work ok!

File: requirements-dev.in [hunk 3]
```diff
@@ -3,7 +3,7 @@ black>=23.7.0
 coverage>=7.2.2
 isort>=5.12.0
 mypy>=1.2.0
-nose>=1.3.7
+nose-py3>=1.6.3 # bug in regular nose here: https://github.com/nose-devs/nose/issues/1099#issuecomment-577647469
```
Comment (bpblanken) [inline requirements-dev.in:6]: I think we shouldn't use `nose` anymore [anyways](https://github.com/broadinstitute/seqr-loading-pipelines/issues/600)

File: hail_scripts/reference_data/clinvar.py [hunk 4]
```diff
@@ -156,8 +157,7 @@ def _parse_clinvar_release_date(local_vcf_path: str) -> str:
     with gzip.open(local_vcf_path, 'rt') as f:
         for line in f:
             if line.startswith('##fileDate='):
-                clinvar_release_date = line.split('=')[-1].strip()
```
Comment (bpblanken) [inline hail_scripts/reference_data/clinvar.py:159]: New version of `ruff` is more aggressive about these, which I think is a good think!

File: .github/workflows/unit-tests.yml [hunk 5]
```diff
@@ -34,7 +34,7 @@ jobs:
       - name: Check Black
         run: black . --check --diff
       - name: Check Ruff
-        run: ruff . --format github
+        run: ruff . --output-format github
```
Comment (bpblanken) [inline .github/workflows/unit-tests.yml:37]: `ruff` has a formatter built in now, which potentially removes the need for `black`.

**Review by hanars — APPROVED** [review]

**Pull request merged**